src/www/vpn_openvpn_server.php

```diff
@@ -202,6 +202,21 @@
 
         if ($result = openvpn_validate_cidr($pconfig['tunnel_network'], gettext('IPv4 Tunnel Network'), false, 'ipv4')) {
             $input_errors[] = $result;
+        } elseif (isset($pconfig['tunnel_network'])) {
```

Comment:
better to use `!empty()`, I think it's always set, check for existence of / makes it a bit more safe.

Reply:
sorry. code works  but i'm obviously confused a little about isset() vs. empty() vs. is_null()

Reply:
I can understand that completely, `isset()` validates if an element exists (the value doesn't matter), `is_null()` (which I personally never use) is more or less the slower version of `$myvar === null` (evaluates to null and the type matches). `empty()` returns true if either the element doesn't exist (same as `!isset()`) or it evaluates to something empty (0, null, "", false)


PHP can be confusing...

Reply:
thanks! I'd better test and not guess from the man )